core/src/dird/ua_restore.cc

```
@@ -721,18 +814,17 @@ static int UserSelectJobidsOrFiles(UaContext* ua, RestoreContext* rx)
-      case 5: /* select backup at specified time */
-        if (!have_date) {
-          if (!get_date(ua, date, sizeof(date))) { return 0; }
-        }
+      case 4: /* Select the most recent backups */ {
+        decltype(date) current_date;
+        bstrutime(current_date, sizeof(current_date), current_time);
+        if (!SelectBackupsBeforeDate(ua, rx, current_date)) { return 0; }
+      } break;
+      case 5: /* select backup at specified time */ {
+        decltype(date) that_date;
+        if (!get_date(ua, that_date, sizeof(that_date))) { return 0; }
```

Cppcheck (2.19.0)

[error] Uninitialized variable

(uninitvar)